### ISSUE TITLE
Switch statement

### ISSUE THREAD

**Issue by anishcana** [anishcana/jovial#6]
#include <iostream>
#include <cstdint>

void processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            break;
        case 2:
            std::cout << "You chose option 2.";
            break;
        case 3:
            std::cout << "You chose option 3.";
            break;
        default:
            std::cout << "Invalid choice.";
    }
}

int main() {
    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    processChoice(choice);

    return 0;
}

**Comment by anishcana** [comment]
#include <iostream>
#include <cstdint>

int processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            return 1;
        case 2:
            std::cout << "You chose option 2.";
            return 2;
        case 3:
            std::cout << "You chose option 3.";
            return 3;
        default:
            std::cout << "Invalid choice.";
            return -1; // Indicate invalid choice
    }
}

int main() {
    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    int selectedOption = processChoice(choice);

    if(selectedOption == -1) {
        // Handle invalid choice
        return 1;
    }

    // Do something with the selectedOption if needed

    return 0;
}

**Comment by anishcana** [comment]
#include <iostream>
#include <cstdint>

enum class Mode {
    Mode1,
    Mode2,
    Mode3,
    Invalid
};

Mode processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            return Mode::Mode1;
        case 2:
            std::cout << "You chose option 2.";
            return Mode::Mode2;
        case 3:
            std::cout << "You chose option 3.";
            return Mode::Mode3;
        default:
            std::cout << "Invalid choice.";
            return Mode::Invalid;
    }
}

int main() {
    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    Mode selectedMode = processChoice(choice);

    switch(selectedMode) {
        case Mode::Mode1:
            // Do something for Mode 1
            break;
        case Mode::Mode2:
            // Do something for Mode 2
            break;
        case Mode::Mode3:
            // Do something for Mode 3
            break;
        case Mode::Invalid:
            // Handle invalid choice
            return 1;
    }

    return 0;
}

**Comment by anishcana** [comment]
#include <iostream>
#include <cstdint>

struct ModeState {
    std::uint8_t state;

    // Constructor to initialize state
    ModeState(std::uint8_t initialState) : state(initialState) {}

    // Function to update state
    void updateState(std::uint8_t newState) {
        state = newState;
    }
};

enum class Mode {
    Mode1 = 1,
    Mode2 = 2,
    Mode3 = 3,
    Invalid = -1
};

Mode processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            return Mode::Mode1;
        case 2:
            std::cout << "You chose option 2.";
            return Mode::Mode2;
        case 3:
            std::cout << "You chose option 3.";
            return Mode::Mode3;
        default:
            std::cout << "Invalid choice.";
            return Mode::Invalid;
    }
}

int main() {
    // Define ModeState objects for each mode
    ModeState mode1State(0b00000001); // Example initial state for Mode 1
    ModeState mode2State(0b00000010); // Example initial state for Mode 2
    ModeState mode3State(0b00000100); // Example initial state for Mode 3

    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    Mode selectedMode = processChoice(choice);

    switch(selectedMode) {
        case Mode::Mode1:
            // Access and modify state for Mode 1
            mode1State.updateState(0b00001000); // Example state update for Mode 1
            break;
        case Mode::Mode2:
            // Access and modify state for Mode 2
            mode2State.updateState(0b00010000); // Example state update for Mode 2
            break;
        case Mode::Mode3:
            // Access and modify state for Mode 3
            mode3State.updateState(0b00100000); // Example state update for Mode 3
            break;
        case Mode::Invalid:
            std::cout << " Invalid choice.";
            return 1;
    }

    return 0;
}

**Comment by anishcana** [comment]
#include <iostream>
#include <cstdint>

// Define states for each mode
std::uint8_t mode1State = 0b00000001;
std::uint8_t mode2State = 0b00000010;
std::uint8_t mode3State = 0b00000100;

enum class Mode {
    Mode1 = 1,
    Mode2 = 2,
    Mode3 = 3,
    Invalid = -1
};

Mode processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            return Mode::Mode1;
        case 2:
            std::cout << "You chose option 2.";
            return Mode::Mode2;
        case 3:
            std::cout << "You chose option 3.";
            return Mode::Mode3;
        default:
            std::cout << "Invalid choice.";
            return Mode::Invalid;
    }
}

int main() {
    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    Mode selectedMode = processChoice(choice);

    switch(selectedMode) {
        case Mode::Mode1:
            // Access and modify state for Mode 1
            mode1State |= 0b00001000; // Example state update for Mode 1
            break;
        case Mode::Mode2:
            // Access and modify state for Mode 2
            mode2State |= 0b00010000; // Example state update for Mode 2
            break;
        case Mode::Mode3:
            // Access and modify state for Mode 3
            mode3State |= 0b00100000; // Example state update for Mode 3
            break;
        case Mode::Invalid:
            std::cout << " Invalid choice.";
            return 1;
    }

    return 0;
}

**Comment by anishcana** [comment]
#include <iostream>
#include <cstdint>

// Define states for each mode in hexadecimal representation
std::uint8_t mode1State = 0x01;
std::uint8_t mode2State = 0x02;
std::uint8_t mode3State = 0x04;

enum class Mode {
    Mode1 = 1,
    Mode2 = 2,
    Mode3 = 3,
    Invalid = -1
};

Mode processChoice(std::uint8_t choice) {
    switch(choice) {
        case 1:
            std::cout << "You chose option 1.";
            return Mode::Mode1;
        case 2:
            std::cout << "You chose option 2.";
            return Mode::Mode2;
        case 3:
            std::cout << "You chose option 3.";
            return Mode::Mode3;
        default:
            std::cout << "Invalid choice.";
            return Mode::Invalid;
    }
}

int main() {
    std::uint8_t choice;
    std::cout << "Enter a number (1-3): ";
    std::cin >> choice;

    Mode selectedMode = processChoice(choice);

    switch(selectedMode) {
        case Mode::Mode1:
            // Access and modify state for Mode 1
            mode1State |= 0x08; // Example state update for Mode 1
            break;
        case Mode::Mode2:
            // Access and modify state for Mode 2
            mode2State |= 0x10; // Example state update for Mode 2
            break;
        case Mode::Mode3:
            // Access and modify state for Mode 3
            mode3State |= 0x20; // Example state update for Mode 3
            break;
        case Mode::Invalid:
            std::cout << " Invalid choice.";
            return 1;
    }

    return 0;
}

**Comment by anishcana** [comment]
import org.apache.poi.ss.usermodel.*;

public class ExcelReader {
    public static void main(String[] args) {
        // Assume wb is your Workbook object containing the Excel data
        Sheet sheet = wb.getSheetAt(0);
        for (Row row : sheet) {
            for (Cell cell : row) {
                if (cell.getCellType() == CellType.ERROR) {
                    if (cell.getErrorCellValue() == FormulaError.NA) {
                        // Handle #N/A error
                        System.out.println("Encountered #N/A error in cell " + cell.getAddress());
                    }
                }
            }
        }
    }
}

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html>
<head>
    <title>List of People</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
</head>
<body>

<h2>List of People</h2>
<ul id="peopleList"></ul>

<script>
  // Iterate over the list of people and append them to the ul element
  <% for(Person person : people) { %>
    var name = "<%= person.getName() %>";
    var age = <%= person.getAge() %>;
    var listItem = $("<li>").text(name + ", " + age);
    $("#peopleList").append(listItem);
  <% } %>
</script>

</body>
</html>

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get ID from URL Parameter</title>
<script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
</head>
<body>

<div id="result"></div>

<script>
$(document).ready(function(){
    // Get the ID from the URL
    var id = getIdFromUrl();
    
    // AJAX request
    $.ajax({
        url: 'your_api_endpoint_here',
        method: 'GET',
        data: { id: id }, // Pass the ID as a parameter
        success: function(response){
            // Handle the response
            $('#result').html(response);
        },
        error: function(xhr, status, error){
            // Handle errors
            console.error(xhr.responseText);
        }
    });
});

// Function to get ID from URL
function getIdFromUrl() {
    var url = window.location.href;
    var parts = url.split('/');
    return parts[parts.length - 1]; // Get the last part of the URL
}
</script>

</body>
</html>

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Key-Value Pair Table</title>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
<style>
    table {
        border-collapse: collapse;
        width: 100%;
    }

    th, td {
        border: 1px solid #dddddd;
        text-align: left;
        padding: 8px;
    }
</style>
</head>
<body>

<table id="keyValueTable">
    <thead>
        <tr>
            <th>Key</th>
            <th>Value</th>
        </tr>
    </thead>
    <tbody>
        
    </tbody>
</table>

<script>
    $(document).ready(function() {
        var keyValuePairs = {
            "Name": "John Doe",
            "Age": 30,
            "Email": "john.doe@example.com"
        };

        var tbody = $('#keyValueTable tbody');

        $.each(keyValuePairs, function(key, value) {
            var row = '<tr>' +
                '<td>' + key + '</td>' +
                '<td>' + value + '</td>' +
                '</tr>';
            tbody.append(row);
        });
    });
</script>

</body>
</html>

**Comment by anishcana** [comment]
<table id="keyValueTable">
    <thead>
        <tr>
            <th>Key</th>
            <th>Value</th>
        </tr>
    </thead>
    <tbody>
        
    </tbody>
</table>

**Comment by anishcana** [comment]
table id="keyValueTable">
    <thead>
        <tr>
            <th>Key</th>
            <th>Value</th>
        </tr>
    </thead>
    <tbody>
        
    </tbody>
</table

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Clickable Key-Value Pair Table</title>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
<style>
    table {
        border-collapse: collapse;
        width: 100%;
    }

    th, td {
        border: 1px solid #dddddd;
        text-align: left;
        padding: 8px;
    }

    /* Highlight effect */
    tr:hover {
        background-color: #f5f5f5;
        cursor: pointer;
    }
</style>
</head>
<body>

<table id="keyValueTable">
    <thead>
        <tr>
            <th>Key</th>
            <th>Value</th>
        </tr>
    </thead>
    <tbody>
        
    </tbody>
</table>

<script>
    $(document).ready(function() {
        var keyValuePairs = {
            "Name": "John Doe",
            "Age": 30,
            "Email": "john.doe@example.com"
        };

        var tbody = $('#keyValueTable tbody');

        $.each(keyValuePairs, function(key, value) {
            var row = '<tr>' +
                '<td>' + key + '</td>' +
                '<td>' + value + '</td>' +
                '</tr>';
            tbody.append(row);
        });

        // Handle click event for table rows
        tbody.on('click', 'tr', function() {
            var key = $(this).find('td:first').text();
            var value = $(this).find('td:last').text();
            alert("Clicked row with key: " + key + " and value: " + value);
            // You can perform any action here with the clicked row's key and value
        });
    });
</script>

</body>
</html>

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Clickable Rows Example</title>
    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
    <style>
        table {
            border-collapse: collapse;
            width: 100%;
        }
        th, td {
            border: 1px solid black;
            padding: 8px;
            text-align: left;
        }
        tr:nth-child(even) {
            background-color: #f2f2f2;
        }
        tr:hover {
            background-color: #ddd;
        }
    </style>
</head>
<body>

<table id="myTable">
    <thead>
        <tr>
            <th>Name</th>
            <th>Age</th>
            <th>Email</th>
        </tr>
    </thead>
    <tbody>
        <tr class="row" data-id="1">
            <td>John Doe</td>
            <td>30</td>
            <td>john@example.com</td>
        </tr>
        <tr class="row" data-id="2">
            <td>Jane Smith</td>
            <td>25</td>
            <td>jane@example.com</td>
        </tr>
    </tbody>
</table>

<script>
    $(document).ready(function(){
        $(".row").click(function() {
            var id = $(this).data("id");
            alert("Row with ID " + id + " clicked!");
            // You can replace alert with your custom code
        });
    });
</script>

</body>
</html>

**Comment by anishcana** [comment]
FROM openjdk:17
      VOLUME /tmp
      EXPOSE 8080
      ARG JAR_FILE=target/com-minikube-0.0.1-SNAPSHOT.jar
      ADD ${JAR_FILE} com-minikube-0.0.1-SNAPSHOT.jar
      ENTRYPOINT ["java","-jar","/com-minikube-0.0.1-SNAPSHOT.jar"]
